src/pages/signup.tsx

```diff
@@ -42,11 +43,22 @@ export default function Signup() {
 
   const [serverErrorState, setServerError] = useState<string | null>(null);
 
-  const onSubmitHandler: SubmitHandler<FormValues> = async () => {
+  const router = useRouter();
+
+  const onSubmitHandler: SubmitHandler<FormValues> = async (formData) => {
     try {
       // @TODO Implement submit
-      // await onSubmit(formData.email, formData.password);
+      // await onSubmit(formData.email, formData.password, formData.fullName);
+
+      const signUpPayload = {
+        email: formData.email,
+        password: formData.password,
+        fullName: formData.fullName,
+      };
+
+      axiosSignUp('/api/v1/auth/signup', signUpPayload);
```

Comment:
Same as in the `login` this can be generalized without using a path:

```suggestion
      const { data } = await signUpWithEmailAndPassword(signUpPayload);
```